File: backend/app/schemas/job_application.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Literal, Optional

Status = Literal["saved", "applied", "interviewing", "offer", "rejected", "withdrawn"]

class JobApplicationCreate(BaseModel):
    company: str
    position: str
    job_url: Optional[str] = None
    location: Optional[str] = None
    salary_range: Optional[str] = None
    currency: Optional[str] = "USD"
    status: Status = "saved"
    source: Optional[str] = None
    notes: Optional[str] = None
    applied_date: Optional[date] = None
    description: Optional[str] = None

class JobApplicationOut(JobApplicationCreate):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    match_percentage: Optional[float] = None
    skill_match: Optional[float] = None
    experience_match: Optional[float] = None
    needs_follow_up: Optional[bool] = None
    days_since_update: Optional[int] = None

    class Config:
        from_attributes = True

File: frontend/types/job_application.ts
export type ApplicationStatus =
  | "saved"
  | "applied"
  | "interviewing"
  | "offer"
  | "rejected"
  | "withdrawn";

export interface JobApplication {
  id: number;
  company: string;
  position: string;
  job_url?: string | null;
  location?: string | null;
  salary_range?: string | null;
  currency?: string | null;
  status: ApplicationStatus;
  source?: string | null;
  notes?: string | null;
  description?: string | null;
  applied_date?: string | null;
  created_at: string;
  updated_at?: string | null;
  match_percentage?: number | null;
  skill_match?: number | null;
  experience_match?: number | null;
  needs_follow_up?: boolean | null;
  days_since_update?: number | null;
}
